pkg/codegen/tensorflow/codegen.go

```diff
@@ -282,6 +282,14 @@ func deriveFeatureColumnCodeAndFieldDescs(trainStmt *ir.TrainStmt) (featureColum
 		perTargetFeatureColumnsCode := []string{}
 		for _, fc := range fcList {
 			fcCode, err := codegen.GenerateFeatureColumnCode(fc, "tf")
+			// FIXME(sneaxiy): auto fill embedding layer to categorical column
+			// This cannot be moved to codegen.GenerateFeatureColumnCode, because
+			// XGBoost model does not support embedding feature column
+			if categoryFC, ok := fc.(ir.CategoricalColumn); ok {
+				fcCode = fmt.Sprintf("tf.feature_column.embedding_column(%s, dimension=%d, combiner=\"%s\")",
```

Comment:
`dimension` for `embedding_column` is the width of the embedding table, yet using `categoryFC.NumClass()` is the height of the embedding table.

And, maybe we should not default add `embedding` to all `categorical` columns?

Reply:
Removed.